crates/router/src/core/payments.rs

```diff
@@ -76,6 +78,13 @@ where
         .validate_request(&req, &merchant_account)?;
 
     tracing::Span::current().record("payment_id", &format!("{}", validate_result.payment_id));
+
+    #[cfg(feature = "api_locking")]
+    let acquired_lock =
```

Comment:
api locking should be in server wrap or kind of a middle-ware solution, the solution should not be in `payments_operation_core`.